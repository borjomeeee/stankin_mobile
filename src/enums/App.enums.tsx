export enum NetworkError {
  NONE,
  NOT_CONNECTION,
  WRONG_LOGIN,
  DOWNLOAD_ERROR,
}

export enum AppModalTypes {
  ERROR,
  WARNING,
  PUSH_UP,
}
