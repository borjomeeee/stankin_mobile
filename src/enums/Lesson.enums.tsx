export enum LessonType {
  LECTURE = 'Лекция',
  SEMINAR = 'Семинар',
  LABA = 'Лаба',
  NO_TYPE = 'без типа',
}

export enum LessonGroup {
  NONE = '',
  GROUP_A = 'А',
  GROUP_B = 'Б',
}
