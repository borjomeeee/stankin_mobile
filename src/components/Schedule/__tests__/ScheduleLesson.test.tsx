import React from 'react';
import renderer from 'react-test-renderer';

import {ILesson} from '../../../models/Lesson.model';
import {LessonType, LessonGroup} from '../../../enums/Lesson.enums';

import ScheduleLessonComponent from '../ScheduleLesson.component';
import {shallow} from 'enzyme';

describe('Schedule lesson tests', () => {
  it('Schedule lesson render correctly', () => {
    const lesson: ILesson = {
      id: '1',
      title: 'Технологии программирования',
      type: LessonType.LABA,
      groupOnLesson: LessonGroup.GROUP_A,
      groupId: '1',
      room: '0001',
      teacher: 'Иванов И.И',
      date: '01/01',
      num: 1,
    };

    const tree = renderer
      .create(<ScheduleLessonComponent {...lesson} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Schedule lesson correct view options with full lesson', () => {
    const lesson: ILesson = {
      id: '1',
      title: 'Технологии программирования',
      type: LessonType.LABA,
      groupOnLesson: LessonGroup.GROUP_A,
      groupId: '1',
      room: '0001',
      teacher: 'Иванов И.И',
      date: '01/01',
      num: 1,
    };

    const wrapper = shallow(<ScheduleLessonComponent {...lesson} />);

    expect(wrapper.find('Styled(View)').at(5).find('Styled(Text)').length).toBe(
      2,
    );
    expect(
      wrapper.find('Styled(View)').at(5).find('Styled(Text)').at(0).text(),
    ).toBe(lesson.room);
    expect(
      wrapper.find('Styled(View)').at(5).find('Styled(Text)').at(1).text(),
    ).toBe(lesson.teacher);
  });

  it('Schedule lesson correct view options with one or more incorrect values', () => {
    const lesson: ILesson = {
      id: '1',
      title: 'Технологии программирования',
      type: LessonType.LABA,
      groupOnLesson: LessonGroup.GROUP_A,
      groupId: '1',
      room: '',
      teacher: 'Иванов И.И',
      date: '01/01',
      num: 1,
    };

    const wrapper1 = shallow(<ScheduleLessonComponent {...lesson} />);

    expect(
      wrapper1.find('Styled(View)').at(5).find('Styled(Text)').length,
    ).toBe(1);

    lesson.teacher = '';
    const wrapper2 = shallow(<ScheduleLessonComponent {...lesson} />);

    expect(
      wrapper2.find('Styled(View)').at(5).find('Styled(Text)').length,
    ).toBe(0);
  });
});
