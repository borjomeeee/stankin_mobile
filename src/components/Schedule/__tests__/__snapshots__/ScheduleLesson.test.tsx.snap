// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule lesson tests Schedule lesson render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "stretch",
        "borderColor": "#E4E4E4",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 15,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingRight": 15,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Inter-Bold",
            "fontSize": 14,
          },
        ]
      }
    >
      8:30
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#9E9E9E",
            "fontFamily": "Inter-Bold",
            "fontSize": 14,
          },
        ]
      }
    >
      10:10
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#9E9E9E",
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginTop": 5,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Inter-Bold",
              "fontSize": 12,
            },
          ]
        }
      >
        А
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      Технологии программирования
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 2,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#444444",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Inter-Regular",
                  "marginLeft": 14,
                },
              ]
            }
          >
            0001
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 2,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#444444",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Inter-Regular",
                  "marginLeft": 14,
                },
              ]
            }
          >
            Иванов И.И
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#444444",
              "borderRadius": 3,
              "paddingBottom": 4,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 4,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Inter-Regular",
              },
            ]
          }
        >
          Лаба
        </Text>
      </View>
    </View>
  </View>
</View>
`;
